### DISCUSSION TITLE
12/03 회의록

### DISCUSSION THREAD

**Discussion by seokjin8678** [SCKorea/sc-backend#1] [회의록]
기능 명세서 링크
https://docs.google.com/spreadsheets/d/1dKEjXpjZz2WTIrUKZgQf4u4g1KZvg_IY

국외 서비스(글로벌)을 제공하는 것은 먼 미래 일이라 판단,
따라서 지금은 한국어 서비스를 제공하는 것을 목표

깃허브 Organization "galaxy-korea" 에서 "SCKorea"로 레포지토리 이동
-> 따라서 프론트 레포를 SCKorea에 생성 후 작업해주시면 될 것 같습니다.
-> 네이밍은 "sc-frontend"

패치 코드는 TOTP를 기반으로 생성
-> [사용할 라이브러리 링크](https://github.com/marcelkliemannel/kotlin-onetimepassword)

패쳐 다운로드 조회 수는 Github Release 다운로드 통계 API를 활용

뉴스는 원문을 크롤링 작업을 통해 DB에 저장 (영어)

DB에 저장된 원문을 가공하여 DeepL API를 통해 번역
-> 가공된 원문을 사용하는 이유는 DeepL 비용 때문
-> 가공된 원문에서 텍스트만 필터링하여 리스트로 변환한 뒤 API 요청
-> 순서가 정확하게 보장되니, 순서에 맞춰서 원문의 text를 replace 하면 될 듯
[DeepL API 공식문서 링크](https://www.deepl.com/ko/docs-api/translate-text/translate-text)

번역된 뉴스가 없을 때, 기본적으로 원문을 보여준다.
번역된 뉴스가 있으면, 번역된 원문을 우선적으로 보여준다.

번역된 뉴스가 정확하지 않을 수 있으므로, 수정자 권한을 가진 사용자가 수정할 수 있다.
-> OAuth2 제공자는 Discord 사용 예정

인증 방법은 Session 사용할 지, JWT 사용할 지 고려

**Comment by seokjin8678** [comment]
추후 와이어프레임 설계를 다 같이 하면 좋겠네요.

**Comment by Titancube** [comment]
참석 못해 죄송합니다 ㅠㅠ 와이어프레임 설계 때는 참가하겠습니다.
프론트 레포는 금주 내로 셋업할게요, 본업 쪽이 많이 힘에 부치네요 ㅎㅎ
